bundles/org.openhab.binding.liquidcheck/src/main/java/org/openhab/binding/liquidcheck/internal/LiquidCheckHandler.java

```diff
@@ -0,0 +1,199 @@
+import static org.openhab.binding.liquidcheck.internal.LiquidCheckBindingConstants.RAW_LEVEL_CHANNEL;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.liquidcheck.internal.httpclient.LiquidCheckHttpClient;
+import org.openhab.binding.liquidcheck.internal.json.CommData;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link LiquidCheckHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Goerentz - Initial contribution
+ */
+@NonNullByDefault
+public class LiquidCheckHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LiquidCheckHandler.class);
+
+    private Map<String, String> oldProps;
+
+    private LiquidCheckConfiguration config = getConfigAs(LiquidCheckConfiguration.class);
+    private LiquidCheckHttpClient client = new LiquidCheckHttpClient(config);
+
+    private @Nullable ScheduledFuture<?> polling;
+
+    public LiquidCheckHandler(Thing thing) {
+        super(thing);
+        oldProps = thing.getProperties();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getAsString().contains(MEASURE_CHANNEL)) {
+            if (command instanceof OnOffType) {
+                try {
+                    if (client.isConnected()) {
+                        String response = client.measureCommand();
+                        CommData commandResponse = new Gson().fromJson(response, CommData.class);
+                        if (null != commandResponse) {
+                            if (!"success".equals(commandResponse.context.status)) {
+                                logger.error("Starting the measurement was not successful!");
+                            }
+                        } else {
+                            logger.error("The object commandResponse is null!");
+                        }
+                    }
+                } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
Please catch separately InterruptedException to call `Thread.currentThread().interrupt();` inside the block.